components/prism-odin.js

```diff
@@ -0,0 +1,61 @@
+(function (Prism) {
+	var escapes = /\\(?:["'\\abefnrtv]|0[0-7]{2}|U[\dA-Fa-f]{6}|u[\dA-Fa-f]{4}|x[\dA-Fa-f]{2})/.source;
+
+	function withEscapes(pattern, flags) {
+		return RegExp(pattern.replace(/<escapes>/g, escapes), flags);
+	}
+
+	Prism.languages.odin = {
+		/**
+		 * The current implementation of multiline comments doesn't support nesting.
+		 */
+		'comment': [
+			{
+				pattern: /\/\*[\s\S]*?(?:\*\/|$)/,
+				greedy: true
+			},
+			{
+				pattern: /#![^\n\r]*/,
+				greedy: true
+			},
+			{
+				pattern: /\/\/[^\n\r]*/,
+				greedy: true
+			}
+		],
+
+		/**
+		 * Should be found before strings because of '"'"- and '`'`-like sequences.
+		 */
+		'char': {
+			pattern: withEscapes(/'(?:<escapes>|[^\n\r'\\])'/.source),
```

Comment:
1. This also need to be greedy to behave correctly. You can read more about this here: #3265.
2. Handling escape sequences explicitly like this actually has some disadvantages:
    1. We need to maintain the list of all possible escape sequences.
    2. We can't highlight incomplete or just slightly wrong characters. Highlighting slightly wrong characters is useful because it makes for a better experience while typing.
    For those reason, I would suggest using a simpler pattern for escape sequences like this:

```suggestion
			pattern: /'(?:\\(?:.|[0Uux][0-9A-Fa-f]{1,6})|[^\n\r'\\])'/,
			greedy: true,
```

The idea behind `[0Uux][0-9A-Fa-f]{1,6}` is that this matches any complete and incomplete numerical escape sequence.

Reply:
I need a couple of days to answer.

Reply:
No worries. Take your time. :)

Reply:
1. Done. I’ve made the `char` pattern greedy.
2. Work in progress.